terraform-common.mk

```diff
@@ -0,0 +1,41 @@
+ENV_NAME := $(notdir $(shell cd $(PWD) && pwd))
```

Comment:
There were some bits I wanted to reuse, and others not, so I figured splitting it up was an OK solution :+1: